cirq/ops/probable_gate.py

```diff
@@ -0,0 +1,132 @@
+# Copyright 2018 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numbers
+from typing import Tuple, TYPE_CHECKING, Dict, Any
+
+import numpy as np
+
+from cirq import protocols, value
+from cirq.ops import raw_types
+from cirq._compat import proper_repr
+
+if TYPE_CHECKING:
+    import cirq
+
+
+@value.value_equality(approximate=True)
+class ProbableGate(raw_types.Gate):
+    """Applies a sub operation with some probability."""
+
+    def __init__(self, *, sub_gate: 'cirq.Gate', probability: value.TParamVal):
```

Comment:
why not just `gate`?  I'm not sure what it's `sub` to.

Reply:
It's `sub` to `self`. Same as in `ControlledGate`.

Reply:
Make sense, except since the reader doesn't know it's a channel it's not clear its part of the channel ;) I find `sub` a throw away word, from context it's clear there is only one object necessary for construction of the object.